### DISCUSSION TITLE
Can we have an easy way of writing namespaces instead of changing it every time

### DISCUSSION THREAD

**Discussion by stealthAngel** [dotnet/csharplang#7266] [Q&A]
Imagine we have this folder structure:

Library
--- Books
--- --- BookModel

The namespace is then 

`namespace Library.Books.BookModel;`

However, if we move it to somewhere else we need to change the namespace again.

Can't we do something like this?

`namespace thisfolder;`

so that when the file moves to another location the namespace automatically changes internally?

**Comment by Richiban** [comment]
This is, IMO, better solved with an editor feature rather than the language itself (which currently has no concept of files or folder structures).

I know ReSharper has a quick action to "fix" namespaces in files, and you can even do it across an entire project in one go. Perhaps VS has the same?

**Reply by TahirAhmadov** [reply]
VS 2022 (definitely 17.5) has automatic namespace adjustment. It is just a little wonky in some edge cases, but it works 99% of the time.

**Comment by Captain-Quack** [comment]
As @TahirAhmadov mentioned, VS 2022 can do this. If you move a file to a folder, it should ask you if you want to change the namespace.
